lang/compiler.go

```diff
@@ -215,37 +215,30 @@ type Program struct {
 
 func (p *Program) Start(ctx context.Context, alloc *memory.Allocator) (flux.Query, error) {
 	ctx, cancel := context.WithCancel(ctx)
-	results := make(chan flux.Result)
-	q := &query{
-		results: results,
-		alloc:   alloc,
-		cancel:  cancel,
-		stats: flux.Statistics{
-			Metadata: make(flux.Metadata),
-		},
-	}
-	q.wg.Add(1)
-	go p.run(ctx, q)
-	return q, nil
-}
-
-func (p *Program) run(ctx context.Context, q *query) {
-	defer q.wg.Done()
-	defer close(q.results)
+	q := NewQuery(alloc, cancel)
```

Comment:
I would like it if we didn't publicly expose the implementation of query here so can we revert this section back to what it was?

Reply:
You're right.  I'll revert it.